### ISSUE TITLE
Add possibility to apply comments for coverage settings in transforms

### ISSUE THREAD

**Issue by Hobart2967** [monounity/karma-typescript#152]
Hi, 

I need a way to apply transforms to the transpiled JS.  Example:

Apply "ignore" statements for istanbul to deactivate some parts via regex (IIFE in my case). Files are not bundled.

```
(context: any, callback: Function) => {
    if (context.module.indexOf('.ts') !== context.module.length - '.ts'.length) {
        return callback(undefined, false);
    }

    this._grunt.log.debug(`Transforming file ${context.module}`);
    const iifeRegex = /}\)\((.*\|\|.*)\);/g;
    context.source = context.source.replace(iifeRegex, (str: string, catch1: string, catch2: string) => '})(/* istanbul ignore next */ ' + catch1 + ');');

    return callback(undefined, true)
}                
```

**Comment by erikbarke** [comment]
Hey @Hobart2967, would it be possible to add these comments to the Typescript source code? Or would the comments be misaligned in the transpiled code?

Otherwise, this would require exposing the transpiled and transformed code in a new "post-transform" or "pre-bundling" step.

**Comment by Hobart2967** [comment]
Yes the comments are misaligned after transpiling. That is basically the problem. I have solved it by adding the emitOutput property to the context and calling the callback with "modified = false" to avoid re-transpiling.

# Transformer.js/TS
>           async.eachSeries(bundleQueue, function (queued, onQueueProcessed) {
>             var context = {
>                 config: _this.config,
>                 filename: queued.file.originalPath,
>                 module: queued.file.originalPath,
>                 source: queued.emitOutput.sourceFile.getFullText(),
                   emitOutput: queued.emitOutput,
>                 ts: {
>                     ast: queued.emitOutput.sourceFile,
>                     version: ts.version
>                 }
>             };
>             async.eachSeries(transforms, function (transform, onTransformApplied) {
>                 process.nextTick(function () {
>                     transform(context, function (error, dirty) {
>                         _this.handleError(error, transform, context);
>                         if (dirty) {
>                             var transpiled = ts.transpileModule(context.source, {
>                                 compilerOptions: _this.project.getTsconfig().options,
>                                 fileName: context.filename
>                             });
                             queued.emitOutput.outputText = transpiled.outputText;
                             queued.emitOutput.sourceMapText = transpiled.sourceMapText;
>                         }					
> 						
>                         onTransformApplied();
>                     });
>                 });
>             }, onQueueProcessed);
>         }, onTransformsApplied);

# Usage example

```
    bundlerOptions: {
        transforms: [
            (context: any, callback: Function) => {
                if (!context.emitOutput) {                                        
                    return callback(undefined, false);
                }                                                             

                const regex = /\}\)\(([^)]+\))/g;
                const subst = "})(/* istanbul ignore next */$1";
                let str = context.emitOutput.outputText;
                context.emitOutput.outputText = str.replace(regex, subst);                                        
        
                return callback(undefined, false);
            }                                
        ]                    
    },
```

**Comment by Hobart2967** [comment]
Heya, added a pull request to this issue that should solve it. Checks passed.

**Comment by erikbarke** [comment]
Solved in #153, closing.

**Comment by Hobart2967** [comment]
Cool, can't wait for version 3.0.6!
